contracts/deploy/Create3.sol

```diff
@@ -26,8 +26,11 @@ library Create3 {
 
     function deploy(bytes32 salt, bytes memory bytecode) internal returns (address deployed) {
         deployed = deployedAddress(salt, address(this));
+        bytes32 existingCodeHash = deployed.codehash;
 
-        if (deployed.codehash != bytes32(0)) revert AlreadyDeployed();
+        // https://eips.ethereum.org/EIPS/eip-1052
+        // keccak256('') == 0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470
+        if (existingCodeHash != bytes32(0) && existingCodeHash != 0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470) revert AlreadyDeployed();
```

Comment:
This also needs to be handled during the sanity check at line 44. For e.g. a contract can `selfdestruct` and use it's own address to send the remaining funds, so it's codehash won't be 0. I'd also be open to remove the sanity check on line 44 altogether and rely on line 39 to check for failure.

Reply:
I can add the additional check to line 44

Reply:
Thanks